File: LeetCodeSolutions/LeetCode_0283.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        nums0 = [i for i in nums if not i]
        nums1 = [i for i in nums if i]
        nums[:] = nums1 + nums0

File: LeetCodeSolutions/LeetCode_0283.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
       int insert_idx = 0, cur = 0;
       while(cur < nums.size()){
           if (nums[cur]){
               nums[insert_idx] = nums[cur];
               ++insert_idx;
           }
           ++cur;
       }
       for (; insert_idx < nums.size(); ++insert_idx)
            nums[insert_idx] = 0;
    }
};